### DISCUSSION TITLE
How to configure MultipartParseOptions?

### DISCUSSION THREAD

**Discussion by rogerrohrbach** [falconry/falcon#2019] [Q&A] [answered]
I'd like to set `falcon.media.multipart.MultipartParseOptions.max_body_part_count` to something other than the default value.  There's no documentation for how to do that, and I can't for the life of me figure it out.

**Comment by rogerrohrbach** [comment, marked as answer]
I answered my own question by examining the source:
```python
from falcon.media.multipart import MultipartFormHandler, MultipartParseOptions

options = MultipartParseOptions()
options.max_body_part_count = 0

app = falcon.App()
app.req_options.media_handlers[falcon.MEDIA_MULTIPART] = MultipartFormHandler(options)
```
It's weird you can't set options in the constructor.

**Comment by vytas7** [comment]
Hi @rogerrohrbach!
That probably works too, although otherwise [`MultipartFormHandler`](https://falcon.readthedocs.io/en/stable/api/media.html#falcon.media.MultipartFormHandler) is modeled after `falcon.App`, and you're supposed to let it initialize its own options, then modify them directly on its instance:
```python
>>> import falcon.media
>>> handler = falcon.media.MultipartFormHandler()
>>> handler.parse_options.max_body_part_count
64
>>> handler.parse_options.max_body_part_count = 0
```

You're right that this is not properly documented for `MultipartFormHandler`; I'll create an issue for that.

The only place where this is illustrated is a [recipe](https://falcon.readthedocs.io/en/stable/user/recipes/multipart-mixed.html) explaining how to deal with (legacy) nested multipart forms, definitely not the most obvious place to look for.

**Reply by rogerrohrbach** [reply]
@vytas7 It's actually that recipe (which is in the repo) that clued me in.  But I'll refactor my code to avoid explicitly instantiating `MultipartParseOptions`.   Thanks!

**Comment by vytas7** [comment]
I've now created an issue from this discussion too: #2020.
